wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/vfs/impl/file/event/EventTreeHelper.java

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.vfs.impl.file.event;
+
+import com.google.common.annotations.Beta;
+
+import org.eclipse.che.api.vfs.Path;
+
+import java.util.Optional;
+
+import static java.io.File.separator;
+import static org.eclipse.che.api.vfs.impl.file.event.EventTreeNode.newInstance;
+
+/**
+ * Helper for event tree related operations.
+ *
+ * @see {@link EventTreeNode}
+ *
+ * @author Dmitry Kuleshov
+ *
+ * @since 4.5
+ */
+@Beta
+class EventTreeHelper {
```

Comment:
Is it on purpose that this class is package-private? Does it make sense to use public modifier since some components may add their own event subscribers?